ocs_ci/ocs/cluster.py

```diff
@@ -4008,3 +4008,42 @@ def is_lower_requirements():
         "Cluster worker machines use standard or larger instance types. No lower requirements detected."
     )
     return False
+
+
+def wait_for_percent_used_capacity_reached(
+    expected_used_capacity, timeout=1800, sleep=20
+):
+    """
+    Wait until the used capacity percentage reaches or exceeds a specified threshold.
+
+    This function repeatedly samples the current used capacity using
+    `get_percent_used_capacity()` until it meets or exceeds the `expected_used_capacity`
+    or until the timeout is reached.
+
+    Args:
+        expected_used_capacity (int or float): The percentage of used capacity to wait for.
+        timeout (int): Maximum time to wait in seconds. Defaults to 1800 seconds (30 minutes).
+        sleep (int): Time to wait between checks in seconds. Defaults to 20 seconds.
+
+    Raises:
+        TimeoutExpiredError: If the expected capacity is not reached within the timeout.
+    """
+    logger.info(f"Wait for the percent used capacity to reach {expected_used_capacity}")
+
+    try:
+        for used_capacity in TimeoutSampler(
+            timeout=timeout,
+            sleep=sleep,
+            func=get_percent_used_capacity,
+        ):
+            logger.info(f"Current percent used capacity = {used_capacity}%")
+            if used_capacity >= expected_used_capacity:
+                logger.info(
+                    f"The expected percent used capacity {expected_used_capacity}% reached"
+                )
+                break
+    except TimeoutExpiredError:
```

Comment:
you raise same exception within the exception. you can do raise {message} instead, or if u want to keep it that way, add `from`